WcaOnRails/app/views/registrations/edit_registrations.html.erb

```diff
@@ -9,16 +9,17 @@
       <% else %>
         <h2>Approved registrations</h2>
       <% end %>
-      <% registrations = @competition.registrations.send(status).order(created_at: :asc, id: :asc) %>
-      <%= wca_table table_class: "registrations-table selectable-rows" do %>
+      <% registrations = @competition.registrations.send(status) %>
+      <%= wca_table table_class: "registrations-table", data: { toggle: "table", sort_name: "registration-date-#{status}"  , select_item_name: "selected_registrations[]", click_to_select: "true" } do %>
         <thead>
           <tr>
-            <th><span></span></th>
+            <th data-checkbox="true"></th>
             <th></th>
-            <th class="wca-id">WCA ID</th>
-            <th class="name">Name</th>
-            <th class="countryId">Citizen of</th>
-            <th class="birthday">Birthday</th>
+            <th class="wca-id" data-sortable="true" data-field="wca-id-<%= status %>">WCA ID</th>
+            <th class="name" data-sortable="true" data-field="name-<%= status %>">Name</th>
+            <th class="countryId" data-sortable="true" data-field="countryId-<%= status %>">Citizen of</th>
+            <th class="registration-date" data-sortable="true" data-field="registration-date-<%= status %>">Registrated</th>
+            <th class="birthday" data-sortable="true" data-field="birthday-<%= status %>">Birthday</th>
```

Comment:
Oh weird. I see now that you did that with all the other fields as well. It looks like the problem is with this line from your workaround:

```javascript
  $('table').on('sort.bs.table', function(e, name, order) {
    // The table column that we are sorting by
    var field = $('table').find('th[data-field="' + name + '"] .sortable');
```

See how that just looks for *any* old table, not necessarily the table that we just clicked on? If I change that line to:

```javascript
    var field = $(e.target).find('th[data-field="' + name + '"] .sortable');
```

it seems to behave nicely. Fortunately, I believe this isn't a problem in https://github.com/wenzhixin/bootstrap-table/pull/2145 =)